src/workerd/api/node/tests/util-nodejs-test.js

```diff
@@ -4894,3 +4943,913 @@ export const testTypes = {
+  },
+};
+
+// parallel/test-util-stripvtcontrolcharacters.js
+export const testStripVtControlCharacters = {
+  async test() {
+    // Ref: https://github.com/chalk/ansi-regex/blob/main/test.js
+    const tests = [
+      // [before, expected]
+      [
+        '\u001B[0m\u001B[4m\u001B[42m\u001B[31mfoo\u001B[39m\u001B[49m\u001B[24mfoo\u001B[0m',
+        'foofoo',
+      ], // Basic ANSI
+      ['\u001B[0;33;49;3;9;4mbar\u001B[0m', 'bar'], // Advanced colors
+      ['foo\u001B[0gbar', 'foobar'], // Clear tabs
+      ['foo\u001B[Kbar', 'foobar'], // Clear line
+      ['foo\u001B[2Jbar', 'foobar'], // Clear screen
+    ];
+
+    for (const ST of ['\u0007', '\u001B\u005C', '\u009C']) {
+      tests.push(
+        [`\u001B]8;;mailto:no-replay@mail.com${ST}mail\u001B]8;;${ST}`, 'mail'],
+        [
+          `\u001B]8;k=v;https://example-a.com/?a_b=1&c=2#tit%20le${ST}click\u001B]8;;${ST}`,
+          'click',
+        ]
+      );
+    }
+
+    for (const [before, expected] of tests) {
+      assert.strictEqual(stripVTControlCharacters(before), expected);
+    }
+  },
+};
+
+export const testStyleText = {
+  async test() {
+    const styled = '\u001b[31mtest\u001b[39m';
+    const noChange = 'test';
+
+    [undefined, null, false, 5n, 5, Symbol(), () => {}, {}].forEach(
+      (invalidOption) => {
+        assert.throws(
+          () => {
+            styleText(invalidOption, 'test');
+          },
+          {
+            code: 'ERR_INVALID_ARG_VALUE',
+          }
+        );
+        assert.throws(
+          () => {
+            styleText('red', invalidOption);
+          },
+          {
+            code: 'ERR_INVALID_ARG_TYPE',
+          }
+        );
+      }
+    );
+
+    assert.throws(
+      () => {
+        styleText('invalid', 'text');
+      },
+      {
+        code: 'ERR_INVALID_ARG_VALUE',
+      }
+    );
+
+    assert.strictEqual(
+      styleText('red', 'test', { validateStream: false }),
+      '\u001b[31mtest\u001b[39m'
+    );
+
+    assert.strictEqual(
+      styleText(['bold', 'red'], 'test', { validateStream: false }),
+      '\u001b[1m\u001b[31mtest\u001b[39m\u001b[22m'
+    );
+
+    assert.strictEqual(
+      styleText(['bold', 'red'], 'test', { validateStream: false }),
+      styleText('bold', styleText('red', 'test', { validateStream: false }), {
+        validateStream: false,
+      })
+    );
+
+    assert.throws(
+      () => {
+        styleText(['invalid'], 'text');
+      },
+      {
+        code: 'ERR_INVALID_ARG_VALUE',
+      }
+    );
+
+    assert.throws(
+      () => {
+        styleText('red', 'text', { stream: {} });
+      },
+      {
+        code: 'ERR_INVALID_ARG_TYPE',
+      }
+    );
+
+    // does not throw
+    styleText('red', 'text', { stream: {}, validateStream: false });
+
+    assert.strictEqual(
+      styleText('red', 'test', { validateStream: false }),
+      styled
+    );
+
+    assert.strictEqual(styleText('none', 'test'), 'test');
+
+    // const fd = common.getTTYfd();
```

Comment:
If we're going to implement this, can you add a TODO? If not, we should remove this